From 49bdf4bdccc086ef7bb6fd67876716f3dfe840bb Mon Sep 17 00:00:00 2001
From: Juergen Beisert <jbe@pengutronix.de>
Date: Mon, 22 Feb 2010 21:25:11 +0100
Subject: [PATCH 50/59] imx: postpone framebuffer init until platform devices are there

The imx framebuffer driver needs data from platform devices, so we have
to wait until they are available by specifying __devinit.

Signed-off-by: Juergen Beisert <jbe@pengutronix.de>
---
 drivers/video/imxfb.c |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

Index: linux-2.6.33/drivers/video/imxfb.c
===================================================================
--- linux-2.6.33.orig/drivers/video/imxfb.c
+++ linux-2.6.33/drivers/video/imxfb.c
@@ -686,7 +686,7 @@ static int imxfb_resume(struct platform_
 #define imxfb_resume	NULL
 #endif
 
-static int __init imxfb_init_fbinfo(struct platform_device *pdev)
+static int __devinit imxfb_init_fbinfo(struct platform_device *pdev)
 {
 	struct imx_fb_platform_data *pdata = pdev->dev.platform_data;
 	struct fb_info *info = dev_get_drvdata(&pdev->dev);
@@ -735,7 +735,7 @@ static int __init imxfb_init_fbinfo(stru
 	return 0;
 }
 
-static int __init imxfb_probe(struct platform_device *pdev)
+static int __devinit imxfb_probe(struct platform_device *pdev)
 {
 	struct imxfb_info *fbi;
 	struct fb_info *info;
@@ -921,6 +921,7 @@ void  imxfb_shutdown(struct platform_dev
 
 static struct platform_driver imxfb_driver = {
 	.suspend	= imxfb_suspend,
+	.probe		= imxfb_probe,
 	.resume		= imxfb_resume,
 	.remove		= __devexit_p(imxfb_remove),
 	.shutdown	= imxfb_shutdown,
@@ -957,7 +958,7 @@ int __init imxfb_init(void)
 	if (ret < 0)
 		return ret;
 
-	return platform_driver_probe(&imxfb_driver, imxfb_probe);
+	return platform_driver_register(&imxfb_driver);
 }
 
 static void __exit imxfb_cleanup(void)
